cmd/kubelet/app/options/options.go

```diff
@@ -168,6 +168,10 @@ type KubeletFlags struct {
 	// NodeStatusMaxImages caps the number of images reported in Node.Status.Images.
 	// This is an experimental, short-term flag to help with node scalability.
 	NodeStatusMaxImages int32
+	// PreferredProbeIPFamily is preferred IP address family based on which one of pod IPs is selected
+	// for pod's health, liveness and readiness probes if pod has both IPv4 and IPv6 addresses
+	// Valid options are "ipv4", "ipv6".
+	PreferredProbeIPFamily string
```

Comment:
If we have a concept of "primary" IP family (which I though was agreed upon?), why do we need this?  Are we going to allow pods to request which IP family they want probes on?  Are we going to allow probes on both/all IPs?

Or is this simply plumbing - getting the "primary" decision down to kubelet?

Reply:
Maybe we don't need this knob at this time. Yes, there is a concept of a "primary" IP family (i.e. the family that is used for service IPs). Kubelet *should* be able to determine what that primary IP family is when it's setting up a probe by inspecting the default IP (the singular "podIP") of the pod, since the default IP should match the primary IP family. The only argument for keeping this knob might be to prepare for a future option of using both/all IPs. My apologies if we don't need this, this knob probably should have been deleted from the KEP when we switched from dual-stack service IPs to single-family service IPs.